api/rest.go

```diff
@@ -1585,6 +1608,29 @@ func handleUpdateCollection(cp utils.ContextProvider, exec boil.Executor, c *Par
 	return handleGetCollection(cp, exec, c.ID)
 }
 
+func handleCollectionContentUnitsPosition(exec boil.Executor, id int64) (*events.Event, *HttpError) {
+	collection, err := models.FindCollection(exec, id)
+	event := events.CollectionContentUnitsChangeEvent(collection)
+	if err != nil {
+		return &event, NewInternalError(err)
+	}
+	ccus, errF := models.CollectionsContentUnits(exec,
```

Comment:
I had this sql snippet used manually many times. Maybe use it here ?
It does the whole thing in 1 DB call.

```sql
UPDATE collections_content_units
SET position = s.row_number
FROM (select cu.id, cu.properties ->> 'film_date', ccu.position, ROW_NUMBER()
    OVER (
      ORDER BY cu.properties ->> 'film_date') as row_number
      from collections_content_units ccu
             inner join content_units cu on ccu.content_unit_id = cu.id and ccu.collection_id = ?
      order by cu.properties ->> 'film_date') AS s
WHERE content_unit_id = s.id
  and collection_id = ?;
```

Reply:
fixed
I was think that if it's on one transaction it same as single request